Change installation directory and add dotnet-quality input

IvanZosimov opened the pull request:
**Description:**
In the scope of this PR, installation directories for Linux and Windows were
changed as well as the logic for resolving `dotnet-version` was updated
and a new input `dotnet-quality` was added.

**Related issue:**
https://github.com/actions/runner-images-internal/issues/4058

**Check list:**
- [x] Mark if documentation changes are required.
- [ ] Mark if tests were added or updated to cover the changes.

dmitry-shibanov commented on README.md:
````diff
@@ -114,52 +114,58 @@ steps:
-# It's only the way to push a package to nuget.org feed for macOS/Linux machines due to API key config store limitations.
-- uses: actions/setup-dotnet@v2
+### nuget.org
+```yml
````
Do we need to delete hint about authentication ?

IvanZosimov replied:
It just looks like it's deleted, but it's in line 140 under the `>**Note**` tag.

dmitry-shibanov commented on __tests__/installer.test.ts:
```diff
@@ -1,33 +1,48 @@
+      await io.rmRF(`${toolDir}/*`);
+      await io.rmRF(`${tempDir}/*`);
+    } catch (err) {
+      console.log(err.message);
+      console.log('Failed to remove test directories');
```
I think we can combine them.

dmitry-shibanov commented on __tests__/installer.test.ts:
```diff
@@ -1,33 +1,48 @@
+      await io.rmRF(`${toolDir}/*`);
+      await io.rmRF(`${tempDir}/*`);
+    } catch (err) {
+      console.log(err.message);
       console.log('Failed to remove test directories');
```
Here is the same

e-korolevskii commented on src/installer.ts:
```diff
@@ -2,187 +2,159 @@
+          httpClient,
+          [major, minor]
+        );
+      }
```
```suggestion
if (this.testTag(major)) {
        this.resolvedArgument.type = "channel";
        if (this.testTag(minor)) {
          this.resolvedArgument.value = `${major}.${minor}`;
        } else {
          this.resolvedArgument.type = "channel";
          const httpClientOptions = {
            allowRetries: true,
            maxRetries: 3,
          };
          const httpClient = new hc.HttpClient(
            "actions/setup-dotnet",
            [],
            httpClientOptions
          );
          this.resolvedArgument.value = await this.getReleasesJsonUrl(
            httpClient,
            [major, minor]
          );
        }
      }
```

IvanZosimov replied:
Thanks, good suggestion! Looks better now 😉

Commit:
Implement proposal stated in ADR for setup-dotnet v3

In the scope of this commit such functionalty was added:
1. Installation directories for Windows and Ubuntu were changed to
the folder with preinstalled .NET SDK's
2. Inner logic of interaction with MS dotnet installer scripts was
 changed
3. New input "dotnet-quality" was added
Such functionality was cut out:
1. Input "include-prerelease"

Commit:
Update unit and e2e tests

Commit:
Update documentation

Commit:
Update bash dotnet-installer script

IvanZosimov commented:
Close this draft PR as actual was created https://github.com/actions/setup-dotnet/pull/315
